needle/modules/storage/data/keychain_dump.py

```diff
@@ -32,23 +35,54 @@ def module_pre(self):
     # ==================================================================================================================
     def module_run(self):
         # Composing the command string
-        cmd = '{}'.format(self.device.DEVICE_TOOLS['KEYCHAINDUMPER'])
+        cmd = '{}'.format(self.device.DEVICE_TOOLS['KEYCHAIN_DUMP'])
         msg = "Dumping the keychain"
         if self.options['filter']:
-            cmd += ' | grep -i "{}" -A 3 -B2'.format(self.options['filter'])
+            grep_args = ' -i -a "{}" -C 10 '.format(self.options['filter'])
             msg += ' with filter: %s' % self.options['filter']
-        # Dump Keychain
+        # Dump Keychain (outputs .plist files)
         self.printer.info(msg)
-        out = self.device.remote_op.command_blocking(cmd)
-
-        # Check output
-        if out and filter(lambda x: "README" not in x, out):
-            # Save to file
-            outfile = self.options['output'] if self.options['output'] else None
-            # Print to console
-            self.print_cmd_output(out, outfile)
-        else:
-            if self.options['filter']:
-                self.printer.warning('No content matches the filter. Ensure the screen is unlocked before dumping the keychain')
-            else:
-                self.printer.warning('No content found. Ensure the screen is unlocked before dumping the keychain')
+        self.device.remote_op.command_blocking(cmd)
+
+        # Used to check if any files were found
+        files_exist = False
+
+        # Used to check if there was any file content after filtering
+        file_content = False
+
+        # For each of the four files which keychain_dump outputs
+        for filename in ['cert.plist','genp.plist','inet.plist','keys.plist']:
+
+            # If this file exists
+            if self.device.remote_op.file_exist(filename):
+
+
+                files_exist = True
+
+                # read in the file either with or without the filter
+                if self.options['filter']:
+                    text = self.device.remote_op.read_file(filename,grep_args)
+                else:
+                    text = self.device.remote_op.read_file(filename)
+
+                # If the file (or filtered file) is not empty
+                if text != []:

```

Comment:
but originally the dump wasn't stored in 4 different plist files on the device's disk. If now we have to deal with those files, we might just retrieve them all (so the user can still analyze them manually later if he wants to)